github/Gist.py

```diff
@@ -214,106 +185,75 @@ def create_comment(self, body):
         headers, data = self._requester.requestJsonAndCheck("POST", f"{self.url}/comments", input=post_parameters)
         return github.GistComment.GistComment(self._requester, headers, data, completed=True)
 
-    def create_fork(self):
+    def create_fork(self) -> Gist:
         """
         :calls: `POST /gists/{id}/forks <https://docs.github.com/en/rest/reference/gists>`_
-        :rtype: :class:`github.Gist.Gist`
         """
         headers, data = self._requester.requestJsonAndCheck("POST", f"{self.url}/forks")
         return Gist(self._requester, headers, data, completed=True)
 
-    def delete(self):
+    def delete(self) -> None:
         """
         :calls: `DELETE /gists/{id} <https://docs.github.com/en/rest/reference/gists>`_
-        :rtype: None
         """
         headers, data = self._requester.requestJsonAndCheck("DELETE", self.url)
 
-    def edit(self, description=github.GithubObject.NotSet, files=github.GithubObject.NotSet):
+    def edit(self, description: Opt[str] = NotSet, files: Opt[dict[str, InputFileContent | None]] = NotSet) -> None:
         """
         :calls: `PATCH /gists/{id} <https://docs.github.com/en/rest/reference/gists>`_
-        :param description: string
-        :param files: dict of string to :class:`github.InputFileContent.InputFileContent`
-        :rtype: None
         """
-        assert description is github.GithubObject.NotSet or isinstance(description, str), description
-        assert files is github.GithubObject.NotSet or all(
+        assert is_optional(description, str), description
+        # limitation of `TypeGuard`
+        assert (not is_defined(files)) or all(
```

Comment:
`is_undefined` here can't narrow `files` to `dict`, so  we have to use `not isdefined(files)` or `isinstance(files, _NotSetType)`